### ISSUE TITLE
running zathura with tabbed causes default binds to break

### ISSUE THREAD

**Issue by sebastinas** [pwmt/zathura#432]
*On GitLab by @tokisuno on Dec 7, 2023, 16:51*

---

If I launch zathura wrapped in devour, ^u and ^d work fine. Typically if I run tabbed zathura, I use a function called "tab" in my fish config.fish, which just executes:

```tabbed
tabbed -c zathura {$argv} -e
```

The same behaviour happens when I type \`tabbed -c zathura \* -e\` in the directory with all my pdfs.

**Comment by sebastinas** [comment]
*On GitLab by @jonathannerat on Dec 15, 2023, 21:22*

---

Same things happens to me. I have a .desktop file with `Exec=tabbed -c -r 2 zathura -e "" %U`. Whenever I open any document looks like the mappings work (like `o` to open a document, or `/something` to search for something), but nothings happens until I switch to another tab and come back. Seems like rendering is stuck inside tabbed.

**Comment by sebastinas** [comment]
*On GitLab by @adrianfd on Dec 19, 2023, 22:53*

---

Exactly the same. I have to switch to fullscreen mode in order zathura to be responsive. Freezes when launching with tabbed

**Comment by sebastinas** [comment]
*On GitLab by @RyanC1098 on Dec 20, 2023, 02:49*

---

If I understand the initial bug report correctly, then I am experiencing the same problem. Tabbed works fine with other applications (e.g. st), Zathura works fine when not run inside Tabbed. But as of a week the two do not play well together. Some commands, like scrolling up and down, do work; but others only work if I switch back and forth between tabs, or if I wait for about 60 seconds. Even when opening a file there is about a 60 second delay. I tried rolling back to previous versions of Zathura, but the problem persists.

**Comment by sebastinas** [comment]
*On GitLab by @caleb-lindgren on Feb 12, 2024, 22:44*

---

Experiencing the same bug as described by @jonathannerat. ~~Downgrading to zathura 0.5.3 fixed it.~~

**Comment by sebastinas** [comment]
*On GitLab by @xezav on Mar 23, 2024, 22:14*

---

I have the same issue as @jonathannerat. Press `/` to try searching, nothing happens and the pdf freezes, but if I tab to a different pdf and come back, the original pdf is no longer frozen and the search can proceed.

I don't use a desktop file, instead a script based on zathura-tabbed from https://web.archive.org/web/20180927142054/http://taitran.ca/vim/latex/markdown/2015/03/11/vim-latex-and-markdown-preview-scripts.html

I thought it may be possible to use feedkeys to try debugging this, but it creates a segfault when I try to bind it.

I tried adding some debug code to tabbed as well, but I had trouble getting it to show what keypresses it was passing or how it was processing them.

The problem does not appear in zathura by itself (untabbed), so maybe this isn't a bug in zathura. Still, zathura is the only program which doesn't behave correctly in tabbed, so hopefully something can be done to fix the integration between these great softwares.

**Comment by MaMiMa81** [comment]
Good Afternoon @sebastinas
Like yourself I too have issues with later versions of zathura (with associated girara & zathura-pdf-poppler) with tabbed.

I use tabbed through the preview-tabbed plugin in nnn file manager & posted an issue on their github project:

[https://github.com/jarun/nnn/issues/1793](url)


I am still using zathura v. 0.5.2 with girara v. 0.4.0 & zathura-pdf-poppler v. 0.3.1 but was wondering if you had any joy debugging:

[https://github.com/pwmt/zathura/issues/432](url)

**Comment by rsvdr** [comment]
I see the same issue, running zathura -l debug FILE.pdf shows that `emit_completed_signal() ` does not occur until tabs (or windows) are switched.

**Comment by MaMiMa81** [comment]
https://github.com/pwmt/girara/issues/49

**Comment by MaMiMa81** [comment]
@sebastinas 
@rsvdr 

Please see here - may be of some help

[https://github.com/pwmt/girara/issues/49#issuecomment-2541631307](url)

[https://github.com/jarun/nnn/issues/1793#issuecomment-2541621428](url)

**Comment by Rhinemann** [comment]
A little late to the party, but also same issue, the binds are either broken or the zathura + tabbed combo is so that bindings don't respond in a reasonable amount of time.

In my case possibly/probably connected to the fact that my system is Wayland, and I try to force tabbed to use XWayland somewhat jankily, which makes using tabbed zathura pretty much impossible.

**Comment by sebastinas** [comment]
So unless there is someone that understand tabbed and how this works together with Gtk, I think it's time to drop the support for it. If people cannot use it, there is no point in pretending that we support it.

**Comment by Rhinemann** [comment]
> So unless there is somewhat that understand tabbed and how this works together with Gtk, I think it's time to drop the support for it. If people cannot use it, there is no point in pretending that we support it.

Yeah, that's fair. Originally seeing that issue I was going to feature request implementing tabs internally, since girara states it supports that. But then I looked through the issues, tabs were rejected before and I went with the next best thing, this.

**Comment by caleb-lindgren** [comment]
In case it's helpful to anyone, this is the workaround I settled on: if you're interested in using tabbed I assume it's also likely that you're using dwm, and if you add the [awesomebar patch](https://dwm.suckless.org/patches/awesomebar/) to dwm it has a nice tabbing functionality that's worked well for me.

**Comment by Rhinemann** [comment]
@caleb-lindgren, sadly I'm actually not. I'm on KDE Plasma, just really loving some suckless-like software. I just also happen to like tabbed pdf vievers because it's a giant screen real-estate saver.

**Comment by sebastinas** [comment]
Embedding support was removed with the port to Gtk 4. So this is "fixed" by removal.
